packages/edit-site/src/components/navigation-sidebar/navigation-panel/menus/content-pages.js

```diff
@@ -1,23 +1,82 @@
 /**
  * WordPress dependencies
  */
-import { __experimentalNavigationMenu as NavigationMenu } from '@wordpress/components';
+import {
+	__experimentalNavigationMenu as NavigationMenu,
+	__experimentalNavigationItem as NavigationItem,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
-import NavigationEntityItems from '../navigation-entity-items';
 import { MENU_CONTENT_PAGES, MENU_ROOT } from '../constants';
+import ContentNavigationItem from '../content-navigation-item';
+import SearchResults from '../search-results';
+import useDebouncedSearch from '../use-debounced-search';
 
 export default function ContentPagesMenu() {
+	const {
+		search,
+		searchQuery,
+		onSearch,
+		isDebouncing,
+	} = useDebouncedSearch();
+
+	const { pages, isResolved } = useSelect(
+		( select ) => {
+			const { getEntityRecords, hasFinishedResolution } = select(
+				'core'
+			);
+			const getEntityRecordsArgs = [
+				'postType',
+				'page',
+				{
+					search: searchQuery,
```

Comment:
Same as my remark for posts below: https://github.com/WordPress/gutenberg/pull/27280/files#r563820929